Plotting/composite_plots.py

```diff
@@ -0,0 +1,211 @@
+    ax2.scatter(np.log10(T_e[non_detect]), logR23[non_detect], marker = '<')
+    if err_bars == True:
+        ax2.errorbar(np.log10(T_e[detect]), logR23[detect], xerr = err_dict['T_e_lowhigh_error'], fmt = '.')
+    ax2.set_xlabel('log($T_e$) (K)')
+    ax2.set_ylabel('log($R_{23}$)')
+    ax2.set_title('$R_{23}$ vs. Temperature')
+    pdf_pages.savefig()
+    
+     
+    #log(O32) vs log(Temperature)
+    fig3, ax3 = plt.subplots()
+    ax3.scatter(np.log10(T_e[detect]), logO32[detect], marker = '.')
+    ax3.scatter(np.log10(T_e[non_detect]), logO32[non_detect], marker = '<')
+    if err_bars == True:
+        ax3.errorbar(np.log10(T_e[detect]), logO32[detect], xerr = err_dict['T_e_lowhigh_error'], fmt = '.')
+    ax3.set_xlabel('log($T_e$) (K)')
+    ax3.set_ylabel('log($O_{32}$)')
+    ax3.set_title('$O_{32}$ vs. Temperature')
+    pdf_pages.savefig()
+    
+    
+    #log(Composite Metallicity) vs log(R23)
+    fig4, ax4 = plt.subplots()
+    ax4.scatter(logR23[detect], com_O_log[detect], marker = '.')
+    ax4.scatter(logR23[non_detect], com_O_log[non_detect], marker = '^')
+    if err_bars == True:
+        ax4.errorbar(logR23[detect], com_O_log[detect], yerr = err_dict['12+log(O/H)_lowhigh_error'], fmt = '.')
+    ax4.set_xlabel('log($R_{23}$)')
+    ax4.set_ylabel('12+log(O/H) $T_e$')
+    ax4.set_title('Composite Metallicity vs. $R_{23}$')
+    pdf_pages.savefig()
+    
+    
+    #log(Composite Metallicity) vs log(O32)
+    fig5, ax5 = plt.subplots()
+    ax5.scatter(logO32[detect], com_O_log[detect], marker = '.')
+    ax5.scatter(logO32[non_detect], com_O_log[non_detect], marker = '^')
+    if err_bars == True:
+        ax5.errorbar(logO32[detect], com_O_log[detect], yerr = err_dict['12+log(O/H)_lowhigh_error'], fmt = '.')
+    ax5.set_xlabel('log($O_{32}$)')
+    ax5.set_ylabel('12+log(O/H) $T_e$')
+    ax5.set_title('Composite Metallicity vs. $O_{32}$')
+    pdf_pages.savefig()
+    
+    
+    #log(Temperature) vs log(Avg Mass)
+    fig6, ax6 = plt.subplots()
+    ax6.scatter(logM_avg[detect], np.log10(T_e[detect]), marker = '.')
+    ax6.scatter(logM_avg[non_detect], np.log10(T_e[non_detect]), marker = 'v')
+    if err_bars == True:
+        ax6.errorbar(logM_avg[detect], np.log10(T_e[detect]), yerr = err_dict['T_e_lowhigh_error'], fmt = '.')
+    ax6.set_xlabel('log($\\frac{M_\star}{M_{\odot}}$)')
+    ax6.set_ylabel('log($T_e$) (K)')
+    ax6.set_title('Temperature vs. Avg Mass')
+    pdf_pages.savefig()
+    
+    
+    #log(Composite Metallicity) vs log(Avg Mass)
+    fig7, ax7 = plt.subplots()
+    ax7.scatter(logM_avg[detect], com_O_log[detect], marker = '.')
+    ax7.scatter(logM_avg[non_detect], com_O_log[non_detect], marker = '^')
+    if err_bars == True:
+        ax7.errorbar(logM_avg[detect], com_O_log[detect], yerr = err_dict['12+log(O/H)_lowhigh_error'], fmt = '.')
+    mass_range = np.arange(8.2, 9.9, 0.05)
+    AM_relation = 8.798 - np.log10(1 + ((10**8.901)/(10**mass_range))**0.640)
```

Comment:
One option is refactoring this to an AM13 function that always gives you the same answer based on the input.